models/aeif_cond_beta_multisynapse.cpp

```diff
@@ -144,84 +153,75 @@ aeif_cond_beta_multisynapse::Parameters_::get( DictionaryDatum& d ) const
   def< double >( d, names::I_e, I_e );
   def< double >( d, names::V_peak, V_peak_ );
   def< double >( d, names::gsl_error_tol, gsl_error_tol );
-  def< int >( d, "n_synapses", num_of_receptors_ );
   def< bool >( d, names::has_connections, has_connections_ );
 }
 
 void
 aeif_cond_beta_multisynapse::Parameters_::set( const DictionaryDatum& d )
 {
-
   updateValue< double >( d, names::V_th, V_th );
   updateValue< double >( d, names::V_peak, V_peak_ );
   updateValue< double >( d, names::t_ref, t_ref_ );
   updateValue< double >( d, names::E_L, E_L );
   updateValue< double >( d, names::V_reset, V_reset_ );
-  updateValue< double >( d, names::E_ex, E_ex );
-  updateValue< double >( d, names::E_in, E_in );
 
   updateValue< double >( d, names::C_m, C_m );
   updateValue< double >( d, names::g_L, g_L );
 
-  std::vector< double > tau_tmp;
-  if ( updateValue< std::vector< double > >( d, names::taus_decay, tau_tmp ) )
-  {
-    if ( tau_tmp.size() < taus_decay.size() && has_connections_ == true )
+  std::vector< double > Erev_tmp, taur_tmp, taud_tmp;
+  bool Erev_flag =
+    updateValue< std::vector< double > >( d, names::E_rev, Erev_tmp );
+  bool taur_flag =
+    updateValue< std::vector< double > >( d, names::taus_rise, taur_tmp );
+  bool taud_flag =
+    updateValue< std::vector< double > >( d, names::taus_decay, taud_tmp );
+  if ( Erev_flag || taur_flag || taud_flag )
+  { // receptor arrays have been modified
+    if ( !Erev_flag || !taur_flag || !taud_flag )
     {
       throw BadProperty(
-        "The neuron has connections, therefore the number of ports cannot be "
-        "reduced." );
+        "The reversal potential, synaptic rise time and synaptic decay time "
+        "arrays must be modified altogether." );
```

Comment:
I changed the error message. Please check if it is ok now, otherwise please suggest a suitable one. I modified the behavior for changes in the arrays according to @Silmathoron indication.